app/bundles/LeadBundle/Entity/LeadRepository.php

```diff
@@ -937,4 +937,22 @@ public function getLeadOwner($ownerId)
 
         return $result;
     }
+
+    /**
+     * Gets Do Not Contact Status for Specified Contact
+     *
+     * @param  integer $id
+     *
+     * @return array|false
+     */
+    public function getDoNotEmail($id)
```

Comment:
This shouldn't be needed if updating this endpoint to support all DNC items across channels. Instead, you can get the lead's DNC records with 
```
/** @var Mautic\LeadBundle\Entity\DoNotContact[] */
$dnc = $this->getModel('lead')->getEntity($leadId)->getDoNotContact();
```